### DISCUSSION TITLE
Use of keyring to load username/password?

### DISCUSSION THREAD

**Discussion by junkyardhokie** [philosowaffle/peloton-to-garmin#148] [Ideas]
Have you considered adding the option to use python's keyring module to save/load the username and password? 

To save the password in keyring you would do the following from a python shell the first time:

`keyring.set_password("garmin.user", "myusername", "<password>")`

Then in your source code you would load the password:
`garmin_password = keyring.get_password("garmin.user", "myusername")`

This would support multiple users and remove the requirement to store the password in plain text.

**Comment by philosowaffle** [comment]
Yes long term I would like to find an alternative to saving the passwords in plaintext.  Just haven't gotten around to it yet.  Considering the project is no longer written in python though, I will likely be exploring a more dotnet way of handling password management at that time.

**Reply by philosowaffle** [reply]
One thing I wanted to share as an immediate upgrade to security is you can choose to set your passwords as [Environment variables](https://github.com/philosowaffle/peloton-to-garmin/wiki/Configuration#environment-variables) instead of in the file. While they will still be set in plain text, it does decrease the attack vector as a bad actor would have to actually gain access to the process and not just the file system.

**Reply by junkyardhokie** [reply]
It's a start but not ideal from a shell history standpoint
